### PR TITLE
[TIMOB-26651][TIMOB-26667] Parity in Ti.Locale.setLangauge

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-26651](https://jira.appcelerator.org/browse/TIMOB-26651)

_**i18n/en/strings.xml**_
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
  <string name="this_is_my_key">this is my value</string>
</resources>
```

_**i18n/es/strings.xml**_
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
  <string name="this_is_my_key">este es mi valor</string>
</resources>
```
_**Resources/app.js**_
```js
Ti.Locale.setLanguage('en-US');
Ti.API.info(Ti.Locale.getCurrentLanguage() + '.eql(en)');

Ti.API.info(Ti.Locale.getString('this_is_my_key') + ' should be "this is my value"');

Ti.Locale.setLanguage('es');
Ti.API.info(Ti.Locale.getCurrentLanguage() + '.eql(es)');
Ti.API.info(Ti.Locale.getString('this_is_my_key') + ' should be "este es mi valor"');

Ti.Locale.setLanguage('en-US');
```

Expected: `getString` should consider locale setting. Note that we saw this issue only on Jenkins so far.

[TIMOB-26667](https://jira.appcelerator.org/browse/TIMOB-26667)

```js
Ti.API.info(Ti.Locale.getString('this_should_not_be_found', null));
Ti.API.info(L('this_should_not_be_found', null));
Ti.API.info(Ti.Locale.getString('this_should_not_be_found', 123));
Ti.API.info(L('this_should_not_be_found', 123));
```

Expected: `this_should_not_be_found` should be printed for all cases

### CODE REVIEW

File: Source/Ti/src/Locale.cpp [hunk 1]
```diff
@@ -116,9 +116,15 @@ namespace TitaniumWindows
...
 			ApplicationLanguages::PrimaryLanguageOverride = TitaniumWindows::Utility::ConvertUTF8String(locale);
+			// Try to reload resources now
+			Windows::ApplicationModel::Resources::Core::ResourceContext::GetForCurrentView()->Reset();
+			Windows::ApplicationModel::Resources::Core::ResourceContext::GetForViewIndependentUse()->Reset();
 			// if it was a valid value (because it got set), assume we can now set our internal language/locale to match
 			currentLocale__ = locale;
 			currentLanguage__ = shortLanguage;
+
+			// Just wanted to wait for resource context to update the language hopefully...
+			std::this_thread::sleep_for(std::chrono::milliseconds(100));
```
Comment (garymathews) [inline Source/Ti/src/Locale.cpp]: Minor but, is this necessary?

**Comment by infosia** [comment]
Can we push this into 8.0.0 if time permits, because this introduces breaking-change on `Ti.Locale.getString`?

File: Source/Ti/src/Locale.cpp (hunk 1 above)
Comment (infosia) [reply Source/Ti/src/Locale.cpp]: Yeah, so we saw the result is delayed and it fails the test ([TIMOB-26651](https://jira.appcelerator.org/browse/TIMOB-26651)). It does not happen locally for me but happens on Jenkins. I ended up adding small sleep and Jenkins seems to pass it.

**Comment by keerthi1032** [comment]
FR Passed. Ti.Locale.setLangauge works as expected.Merging
Test Enviornment:
Operating System
  Name                        = Microsoft Windows 10 Pro
  Version                     = 10.0.17134
  Architecture                = 64bit
Node.js
  Node.js Version             = 8.9.1
  npm Version                 = 5.5.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = 8.0.0 local
Device                             =Windows 10 emulator, lumia 650 dual sim.

**Pull request merged**